src/pudl/transform/ferc1.py

```diff
@@ -655,6 +893,41 @@ class Ferc1TableTransformParams(TableTransformParams):
         UnstackBalancesToReportYearInstantXbrl()
     )
     combine_axis_columns_xbrl: CombineAxisColumnsXbrl = CombineAxisColumnsXbrl()
+    reconcile_table_calculations: ReconcileTableCalculations = (
+        ReconcileTableCalculations()
+    )
+
+    @property
+    def xbrl_factoid_name(self) -> str:
+        """Access the column name of the ``xbrl_factiod``."""
+        return self.merge_xbrl_metadata.on
+
+    @property
+    def rename_dicts_xbrl(self):
+        """Compile all of the XBRL rename dictionaries into an ordered list."""
+        # add the xbrl last bc it happens after the inst/dur renames
+        return [
+            self.rename_columns_ferc1.duration_xbrl,
+            self.rename_columns_ferc1.instant_xbrl,
+            self.rename_columns_ferc1.xbrl,
+        ]
+
+    @property
+    def wide_to_tidy_value_types(self) -> list[str]:
+        """Compile a list of all of the ``value_types`` from ``wide_to_tidy``."""
+        # the values in wide_to_tidy are found at the end of the column names and
+        # extracted, so we need to remove all of the suffixes.
+        wide_to_tidy_value_types = []
+        for wide_to_tidy_stage in json.loads(self.wide_to_tidy.json()).values():
+            if not isinstance(wide_to_tidy_stage, list):
+                wide_to_tidy_stage = [wide_to_tidy_stage]
+            for wide_to_tidy in wide_to_tidy_stage:
+                if wide_to_tidy["value_types"]:
+                    wide_to_tidy_value_types.append(wide_to_tidy["value_types"])
+        wide_to_tidy_value_types = pudl.helpers.dedupe_n_flatten_list_of_lists(
+            wide_to_tidy_value_types
+        )
+        return wide_to_tidy_value_types
```

Comment:
i like the idea of migrating them. in that world when you are inside a table transformer would you access `wide_to_tidy_value_types` by `self.params.wide_to_tidy.wide_to_tidy_value_types`?

Reply:
I think it would look like

```py
# All unique value types from just the XBRL wide-to-tidy parameters (this already exists):
self.params.wide_to_tidy.xbrl.value_types
# All unique value types from just the DBF wide-to-tidy parameters (this already exists):
self.params.wide_to_tidy.dbf.value_types
# All unique value types from both XBRL and DBF wide-to-tidy parameters (need to define this property):
self.params.wide_to_tidy.value_types
```

The method that needs to be defined in `WideToTidy` is probably something like:

```py
@property
def value_types() -> list[str]:
    return list(set(self.xbrl.value_types + self.dbf.value_types))
```

Reply:
okay i migrated two of these properties and cleaned up the wide to tidy value grabber. there was more logic in there because the wtt param can either be a `WideToTidy` or a list of `WideToTidy`.